src/components/Player/index.tsx

```diff
@@ -27,26 +45,83 @@ const Player: FC<{ videoId: string }> = ({ videoId }) => {
 			description: videoTemp.items[0].snippet.description,
 		});
 	}, [videoId]);
+	let clock: IRunningTime | undefined;
+
+	function render({ hours, minutes, seconds }: CountdownRenderProps) {
+		clock = { hours, minutes, seconds };
+		return (
+			<TimerTypography variant="h4">
+				{hours}:{minutes}:{seconds}
+			</TimerTypography>
+		);
+	}
+
+	function updateTime() {
+		if (clock !== undefined) {
+			const hours = convertIntoHours(clock);
+			setTimeAction(hours).then(() => {});
+		}
+	}
+
+	const interval = setInterval(updateTime, 120000);
 
 	useEffect(() => {
-		initialize().then(() => {});
+		initialize().then(() => {
+			getTimerAction().then((data) => {
+				setTime(data);
+			});
+		});
 	}, [initialize]);
+	useEffect(() => {
+		return () => {
+			updateTime();
+			clearInterval(interval);
+		};
+	});
```

Comment:
Move the timer logic to Player page instead of the Player component.
After calculating time on Player page update the Player component and Clock component accordingly.